geoh5py/shared/entity.py

```diff
@@ -210,6 +201,58 @@ def create(cls, workspace, **kwargs):
         )
         return new_object
 
+    @abstractmethod
+    def copy(
+        self,
+        parent=None,
+        copy_children: bool = True,
+        clear_cache: bool = False,
+        **kwargs,
+    ):
+        """
+        Function to copy an entity to a different parent entity.
+
+        :param parent: Target parent to copy the entity under. Copied to current
+            :obj:`~geoh5py.shared.entity.Entity.parent` if None.
+        :param copy_children: (Optional) Create copies of all children entities along with it.
+        :param clear_cache: Clear array attributes after copy.
+        :param kwargs: Additional keyword arguments to pass to the copy constructor.
+
+        :return entity: Registered Entity to the workspace.
+        """
+
+    def copy_from_extent(
+        self,
+        extent: list[float] | ndarray,
```

Comment:
**suggestion (YAGNI):** not necessary to support list as param type

else, would be `list[list[float]]` and requires its own set of unit tests...

Reply:
Ok fine. We can force 2D array, will be less risky